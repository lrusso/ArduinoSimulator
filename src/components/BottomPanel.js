import React from "react"

const BottomPanel = ({ children }) => {
  return <div style={styles.container}>{children}</div>
}

const styles = {
  container: {
    position: "fixed",
    left: 0,
    right: 0,
    bottom: 0,
    height: "251px",
    borderTop: "1px solid #D3D3D3",
    backgroundColor: "#F2F2F2",
    borderLeft: "1px solid #D3D3D3",
    overflowX: "hidden",
  },
}

export default BottomPanel
