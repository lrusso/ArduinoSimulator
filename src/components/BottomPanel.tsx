import React from "react"

interface BottomPanelProps {
  children: React.ReactNode
}

const BottomPanel = ({ children }: BottomPanelProps) => {
  return <div style={styles.container}>{children}</div>
}

const styles: { [key: string]: React.CSSProperties } = {
  container: {
    position: "fixed",
    left: 0,
    right: 0,
    bottom: 0,
    height: "280px",
    borderTop: "1px solid #363636",
    backgroundColor: "#181818",
    borderLeft: "1px solid #363636",
    overflowX: "hidden",
  },
}

export default BottomPanel
