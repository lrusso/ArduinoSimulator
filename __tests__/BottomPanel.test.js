import React from "react"
import renderer from "react-test-renderer"
import BottomPanel from "../src/components/BottomPanel"

describe("BottomPanel", () => {
  test("Rendering 1", () => {
    const tree = renderer
      .create(
        <BottomPanel>
          <></>
        </BottomPanel>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
