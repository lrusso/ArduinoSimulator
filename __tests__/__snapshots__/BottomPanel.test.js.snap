// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomPanel Rendering 1 1`] = `
<div
  style={
    {
      "backgroundColor": "#F2F2F2",
      "borderLeft": "1px solid #D3D3D3",
      "borderTop": "1px solid #D3D3D3",
      "bottom": 0,
      "height": "251px",
      "left": 0,
      "overflowX": "hidden",
      "position": "fixed",
      "right": 0,
    }
  }
/>
`;
